finagle-http2/src/test/scala/com/twitter/finagle/http2/ClientFailUpgradeTest.scala

```diff
@@ -25,7 +25,8 @@ class ClientFailUpgradeTest extends AbstractHttp1EndToEndTest {
     TooLongStream,
     Streaming,
     CloseStream,
-    StreamFixed
+    StreamFixed,
+    NoBodyMessage
```

Comment:
Your suspicion sounds reasonable to me. In that case, we might have _timeout_ error or exception?

> Do the tests work if you remove the 101 test from ClientFailUpgradeTest?

Yup, I think it works without 101 test. Currently, 101 test (finagle-http) also fails for `EndToEndTest` with the following error:
`Some(0) was not empty (AbstractEndToEndTest.scala:949)`
[AbstractEndToEndTest.scala:949](https://github.com/twitter/finagle/pull/521/commits/057ed65fd51669e2c36a0af12300cb7f024c6094#diff-b884af71723a791bed9b01d0b4ece1e9R949)
It looks like Content-Length has the value of 0, which shouldn't happen like we can see from `ResponseConformanceFilterTest`...

@bryce-anderson What do you think about this?

>  Could we restructure the tests to use "featureImplemented" to decide whether to include 101 or not instead of just turning them off for CFUT?

Could you show me some sample code of how to use `featureImplemented` please? It seems like there's no code using it in finagle? :eyes: